### PR TITLE
Do start and initialize service at single step [ECR-3222]

### PR DESCRIPTION
**Pull request by alekseysidorov**

### CODE REVIEW

**Comment by popzxc** [comment]
Regarding the PR idea: I like it. Currently the service has a Z state between the start and initialize: service is already running, but it cannot do anything, since it's not initalized.
And since those methods are called one after another I don't see a reason why not to combine them.

**Comment by alekseysidorov** [comment]
Yes that is correct. In old interface there was sitiuations where service became in incomplete state between `start_service` and `initialize_service` invocations. That lead to fatal errros in some cases.
This PR change workflow to avoid this situations by design.
In other words now service is in correct state all the time.

File: examples/sample_runtime/src/main.rs [hunk 1]
```diff
@@ -96,59 +114,28 @@ impl Runtime for SampleRuntime {
...
-            SampleService {
-                name: spec.name.clone(),
-                ..SampleService::default()
-            },
-        );
-        println!("Starting service: {:?}", spec);
+    /// Starts an existing `SampleService` instance with the specified ID.
+    fn restart_service(&mut self, spec: &InstanceSpec) -> Result<(), ExecutionError> {
+        let instance = self.start_service(spec)?;
+        println!("Starting service {}: {:?}", spec, instance);
```
Comment (aleksuss) [inline examples/sample_runtime/src/main.rs]: Why not log macro ?

File: exonum/src/runtime/dispatcher/mod.rs [hunk 2]
```diff
@@ -435,13 +403,13 @@ impl Dispatcher {
         runtime_changes.push(ApiChange::InstanceAdded(instance.id));
     }
 
-    /// Start a new service instance.
+    /// Start a new previously added service instance.
```
Comment (aleksuss) [inline exonum/src/runtime/dispatcher/mod.rs]:
```suggestion
    /// Restart a new previously added service instance.
```

File: exonum/src/runtime/mod.rs [hunk 3]
```diff
@@ -198,49 +198,32 @@ pub trait Runtime: Send + Debug + 'static {
     /// * Ensure that the deployed artifact has the following information, even if it is empty.
     fn artifact_protobuf_spec(&self, id: &ArtifactId) -> Option<ArtifactProtobufSpec>;
 
-    /// Start a new service instance with the given specification.
+    /// Start previously added service instance with the given specification.
```
Comment (aleksuss) [inline exonum/src/runtime/mod.rs]:
```suggestion
    /// Restart previously added service instance with the given specification.
```

**Pull request merged**